synapse/lib/storm.py

```diff
@@ -2391,6 +2409,192 @@ async def execStormCmd(self, runt, genr):
         async for node, path in subg.run(runt, genr):
             yield node, path
 
+class ViewExecCmd(Cmd):
+    '''
+    Execute a storm query in a different view.
+
+    NOTE: Variables are passed through but nodes are not
+
+    Examples:
+
+        // Move some tagged nodes to another view
+        inet:fqdn#foo.bar $fqdn=$node.value() | view.exec 95d5f31f0fb414d2b00069d3b1ee64c6 { [ inet:fqdn=$fqdn ] }
+    '''
+
+    name = 'view.exec'
+    readonly = True
+
+    def getArgParser(self):
+        pars = Cmd.getArgParser(self)
+        pars.add_argument('view', help='The GUID of the view in which the query will execute.')
+        pars.add_argument('storm', help='The storm query to execute on the view.')
+        return pars
+
+    async def execStormCmd(self, runt, genr):
```

Comment:
To avoid problems of "what the heck happened" I want to log the dynamic query text which is possible to be executed here.